From baf1022234f865ea9a887a395e1c5fb4802a12a2 Mon Sep 17 00:00:00 2001
From: Daniel Kurtz <djkurtz@chromium.org>
Date: Mon, 25 Jul 2011 21:16:23 +0800
Subject: [PATCH 2/7] PARTIAL-BACKPORT: xfree86: remove LocalDeviceRec/Ptr definition.

Two names pointing to the same struct for over 7 years now. Remove the
define, if drivers don't want to change they can always do the typedef
themselves.

Rename all "LocalDevicePtr local" to "InputInfoPtr pInfo".

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Adam Jackson <ajax@redhat.com>
Reviewed-by: Daniel Stone <daniel@fooishbar.org>
(cherry picked from commit cdb5863c9f68548afa0c69e08c697c83f9a5e671)

BACKPORT NOTE: Only backporting xf86Helper.c changes.

Conflicts:

	hw/xfree86/common/xf86Xinput.c
	hw/xfree86/common/xf86Xinput.h
---
 hw/xfree86/common/xf86Helper.c |   10 +++++-----
 hw/xfree86/common/xf86Xinput.h |    8 ++++----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/hw/xfree86/common/xf86Helper.c b/hw/xfree86/common/xf86Helper.c
index f05c4b6..1ef005e 100644
--- a/hw/xfree86/common/xf86Helper.c
+++ b/hw/xfree86/common/xf86Helper.c
@@ -1266,7 +1266,7 @@ xf86DrvMsg(int scrnIndex, MessageType type, const char *format, ...)
 /* Print input driver messages in the standard format of
    <driver>: <device name>: <message> */
 void
-xf86VIDrvMsgVerb(LocalDevicePtr dev, MessageType type, int verb, const char *format,
+xf86VIDrvMsgVerb(InputInfoPtr dev, MessageType type, int verb, const char *format,
 		 va_list args)
 {
     char *msg;
@@ -1278,7 +1278,7 @@ xf86VIDrvMsgVerb(LocalDevicePtr dev, MessageType type, int verb, const char *for
 
 /* Print input driver message, with verbose level specified directly */
 void
-xf86IDrvMsgVerb(LocalDevicePtr dev, MessageType type, int verb, const char *format,
+xf86IDrvMsgVerb(InputInfoPtr dev, MessageType type, int verb, const char *format,
 	       ...)
 {
     va_list ap;
@@ -1290,7 +1290,7 @@ xf86IDrvMsgVerb(LocalDevicePtr dev, MessageType type, int verb, const char *form
 
 /* Print input driver messages, with verbose level of 1 (default) */
 void
-xf86IDrvMsg(LocalDevicePtr dev, MessageType type, const char *format, ...)
+xf86IDrvMsg(InputInfoPtr dev, MessageType type, const char *format, ...)
 {
     va_list ap;
 
@@ -2123,7 +2123,7 @@ xf86IsUnblank(int mode)
 }
 
 void
-xf86MotionHistoryAllocate(LocalDevicePtr local)
+xf86MotionHistoryAllocate(InputInfoPtr pInfo)
 {
-    AllocateMotionHistory(local->dev);
+    AllocateMotionHistory(pInfo->dev);
 }
diff --git a/hw/xfree86/common/xf86Xinput.h b/hw/xfree86/common/xf86Xinput.h
index db720f1..dc7c8d2 100644
--- a/hw/xfree86/common/xf86Xinput.h
+++ b/hw/xfree86/common/xf86Xinput.h
@@ -206,14 +206,14 @@ extern _X_EXPORT InputInfoPtr xf86AllocateInput(InputDriverPtr drv, int flags);
 extern _X_EXPORT InputDriverPtr xf86LookupInputDriver(const char *name);
 extern _X_EXPORT InputInfoPtr xf86LookupInput(const char *name);
 extern _X_EXPORT void xf86DeleteInput(InputInfoPtr pInp, int flags);
-extern _X_EXPORT void xf86MotionHistoryAllocate(LocalDevicePtr local);
-extern _X_EXPORT void xf86IDrvMsgVerb(LocalDevicePtr dev,
+extern _X_EXPORT void xf86MotionHistoryAllocate(InputInfoPtr pInfo);
+extern _X_EXPORT void xf86IDrvMsgVerb(InputInfoPtr dev,
 				      MessageType type, int verb,
 				      const char *format, ...) _X_ATTRIBUTE_PRINTF(4,5);
-extern _X_EXPORT void xf86IDrvMsg(LocalDevicePtr dev,
+extern _X_EXPORT void xf86IDrvMsg(InputInfoPtr dev,
 				  MessageType type,
 				  const char *format, ...) _X_ATTRIBUTE_PRINTF(3,4);
-extern _X_EXPORT void xf86VIDrvMsgVerb(LocalDevicePtr dev,
+extern _X_EXPORT void xf86VIDrvMsgVerb(InputInfoPtr dev,
 				       MessageType type,
 				       int verb,
 				       const char *format,
-- 
1.7.3.1

